pkg/agent/openflow/network_policy.go

```diff
@@ -1313,3 +1336,76 @@ func (c *client) ReassignFlowPriorities(updates map[uint16]uint16, table binding
 	}
 	return nil
 }
+
+func parseDropFlow(flow string) (uint32, types.RuleMetric) {
+	// example format:
+	// table=101, n_packets=9, n_bytes=666, priority=200,ip,reg0=0x100000/0x100000,reg3=0x5 actions=drop
+	segs := strings.Split(flow, ",")
+	m := types.RuleMetric{}
+	pkts, _ := strconv.ParseUint(segs[1][strings.Index(segs[1], "=")+1:], 10, 64)
+	m.Packets = pkts
+	m.Sessions = pkts
+	bytes, _ := strconv.ParseUint(segs[2][strings.Index(segs[2], "=")+1:], 10, 64)
+	m.Bytes = bytes
+	id, _ := strconv.ParseUint(segs[6][strings.Index(segs[6], "0x")+2:strings.Index(segs[6], " ")], 16, 64)
+	return uint32(id), m
+}
+
+func parseAllowFlow(flow string) (uint32, types.RuleMetric) {
+	// example format:
+	// table=101, n_packets=0, n_bytes=0, priority=200,ct_state=-new,ct_label=0x1/0xffffffff,ip actions=goto_table:105
+	segs := strings.Split(flow, ",")
+	m := types.RuleMetric{}
+	pkts, _ := strconv.ParseUint(segs[1][strings.Index(segs[1], "=")+1:], 10, 64)
+	m.Packets = pkts
+	if strings.Contains(segs[4], "+") { // ct_state=+new
+		m.Sessions = pkts
+	}
+	bytes, _ := strconv.ParseUint(segs[2][strings.Index(segs[2], "=")+1:], 10, 64)
+	m.Bytes = bytes
+	idRaw := segs[5][strings.Index(segs[5], "0x")+2 : strings.Index(segs[5], "/")]
+	if len(idRaw) > 8 { // only 32 bits are valid.
```

Comment:
What happens to egress rule if only 32 bits are valid? I see that in ct_label we use 64 bits like below:
>// We use the 0..31 bits of the ct_label to store the ingress rule ID and use the 32..63 bits to store the
	// egress rule ID.

Reply:
Thanks for pointing it out, this should be a bug. I will create a PR to fix it.

Reply:
When matching egress rule, there should always be 8 trailing zero while the length of the matching pattern of ingress rule will never exceed 8.
In other words, if the length of `idRaw` exceeds 8, it must for egress, otherwise it's for ingress.

Reply:
I see.. makes sense. The comment was a bit confusing and misleading.